### PR TITLE
Relay export

### PR DESCRIPTION
**Pull request by gheise**
Add a DAG to export public data from the Flashbots relay

### CODE REVIEW

File: dags/relay_export_dag.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        psql_hook = PostgresHook(postgres_conn_id=POSTGRES_CONN_ID)
+        connection = psql_hook.get_conn()
+        with connection.cursor() as cursor:
+            cursor.execute(f"""
+                SELECT * 
+                FROM mainnet_builder_block_submission 
+                WHERE inserted_at > '{datetime.datetime(2022,10,17)}'
+                LIMIT 1
+            """)
+            result = cursor.fetchall()
```
Comment (Ruteri) [inline dags/relay_export_dag.py:27]: `fetchall` might be too much data to handle all at once for a large backlog, you will need to cut up the results (or the query). Fetch by default fetches a bunch of rows, and that might be enough, or you can use `fetchmany` with an explicit max size.

File: dags/relay_export_dag.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    
+        
+        s3_hook = S3Hook('s3_conn')
+        s3_hook.load_file(filename=f'{datetime.now()}', key=f'{datetime.now()}', bucket_name='airflow-testdata')
```
Comment (Ruteri) [inline dags/relay_export_dag.py:31]: I think the bucket name should be configurable via airflow as it's not necessarily going to be public

Comment (gheise) [reply dags/relay_export_dag.py]: i added the bucket name as a secret through the airflow secrets backend

File: dags/relay_export_dag.py (hunk 1 above)
Comment (gheise) [reply dags/relay_export_dag.py:36]: i will change this once the query is finalized and we're done testing

File: dags/relay_export_dag.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import datetime
+import boto3
+import json
+from airflow import DAG
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from airflow.hooks.S3_hook import S3Hook
+from airflow.operators.python import PythonOperator
+DAG_ID = "relay_export_dag"
+POSTGRES_CONN_ID = "boost_relay_read_replica"
+AWS_REGION = "us-east-2"
```
Comment (Ruteri) [inline dags/relay_export_dag.py:10]: should go to variables

File: dags/relay_export_dag.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+session = boto3.session.Session()
+client = session.client(service_name='secretsmanager',region_name=AWS_REGION)
+secret = json.loads(client.get_secret_value(SecretId=SECRET)["SecretString"])
+
+BUCKET_NAME = secret["bucket_name"]
```
Comment (Ruteri) [inline dags/relay_export_dag.py:16]: should go to variables

[the remark this answers is not in the document]
Comment (gheise) [reply dags/relay_export_dag.py]: Done!

File: dags/relay_export_dag.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        POSTGRES_CONN_ID = Variable.get('mainnet-relay-connection-id')
+
+        psql_hook = PostgresHook(postgres_conn_id=POSTGRES_CONN_ID)
+        connection = psql_hook.get_conn()
+        with connection.cursor() as cursor:
+            cursor.execute(f"""
+                SELECT *
+                FROM mainnet_builder_block_submission
+                WHERE inserted_at > '{dag.start_date}'
+                AND inserted_at < '{dag.start_date - datetime.timedelta(hours=1)}'
```
Comment (Ruteri) [inline dags/relay_export_dag.py:30]: With a daily catchup it should be timedelta 1 day right?

File: dags/relay_export_dag.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            """)
+            result = cursor.fetchall()
+
+        s3_hook = S3Hook('s3_conn')
+        s3_hook.load_file_obj(file_obj=io.BytesIO(f"{result}".encode('UTF-8')), key=f'{datetime.datetime.now()}', bucket_name=BUCKET_NAME)
```
Comment (Ruteri) [inline dags/relay_export_dag.py:35]: `datetime.datetime.now()` should probably be the truncated date of `dag.start_date` for daily files

File: dags/relay_export_dag.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        BUCKET_NAME = Variable.get('relay-data-export-bucket-name')
+        POSTGRES_CONN_ID = Variable.get('mainnet-relay-connection-id')
+
+        psql_hook = PostgresHook(postgres_conn_id=POSTGRES_CONN_ID)
+        connection = psql_hook.get_conn()
+        with connection.cursor() as cursor:
+            cursor.execute(f"""
+                SELECT *
+                FROM mainnet_builder_block_submission
+                WHERE inserted_at > '{dag.start_date}'
```
Comment (metachris) [inline dags/relay_export_dag.py:29]: That query is too expensive without an index on insertedAt. Please ping me with any relay db queries going forward 🙏

Comment (metachris) [reply dags/relay_export_dag.py:29]: I can add an index on insertedAt

**Comment by metachris** [comment]
I'd like to understand what prompted this effort?

There are already bulk data exports in the data API format at https://flashbots-boost-relay-public.s3.us-east-2.amazonaws.com/index.html

It seems this DAG is not needed?

Fwiw, I'm also not sure "SELECT *" would be the right approach, and whether all that data should be public.

Please coordinate with the mev-boost team before any efforts that use the boost relay databases 🙏

**Comment by metachris** [comment]
Proposing to close this PR since that data is already exported by the relay itself.

**Pull request closed without merging**